app/models/user.rb

```diff
@@ -4,4 +4,30 @@ class User < ApplicationRecord
   # :confirmable, :lockable, :timeoutable and :omniauthable
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :trackable, :validatable
+
+
+  def has_membership_application?
+    membership_applications.size > 0 && (membership_applications.select { |app| app.persisted? }).count > 0
+  end
+
+
+  def has_company?
+    has_membership_application? && (membership_applications.select { |app| app.company && app.company.persisted? }).count > 0
+  end
+
+
+  def membership_application
+    has_membership_application? ? membership_applications.last : nil
+  end
+
+
+  def company
+    has_company? ? membership_application.company : nil
+  end
```

Comment:
"We're going with the has_many association right now to allow for the possibility of a user submitting many applications."
Why? I don't remember. They could instead update the one they have. But it is a grander scheme decision that has to be talked about before changing (in case they have x set of skills in one company, and y set of skills with the other company... ) ..